### PR TITLE
Support YAML

### PR DESCRIPTION
**Pull request by tkatox**
Support YAML files instead of JSON.

### CODE REVIEW

**Review by na4da — CHANGES_REQUESTED** [review]
It's confirmed that `papery` command installed by `pip install  -e .` works correctly even YAML-based config and index are applied. Thank you!
I commented about the code appearance.

**Review by na4da — APPROVED** [review]
LGTM. Thanks!

File: papery/data/sample/config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+title: My web site with papery
+url: http://www.example.com
+email: webmaster@example.com
+keywords: papery
+description: Web site with papery
+theme: default
+pages:
+  - file: pages/*.md
+    template: page.tmpl
+    path: /
```
Comment (federkasten) [inline papery/data/sample/config.yaml:11]: Please delete the corresponding JSON.

File: README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@ Papery version 0.1.10
 
 # Papery
 
-papery is  is a simple, static site generator written in Python - supports Markdown/JSON input and Jinja2 templating.
+papery is  is a simple, static site generator written in Python - supports Markdown/JSON/YAML input and Jinja2 templating.
```
Comment (federkasten) [inline README.md:5]:
```suggestion
papery is  is a simple, static site generator written in Python - supports Markdown, YAML, and JSON input and Jinja2 templating.
```

File: papery/engine.py [hunk 3]
```diff
@@ -69,16 +70,26 @@ def __init__(self):
...
-                config_path = 'config.json'
-            elif os.path.exists('.config.json'):
-                config_path = '.config.json'
-
-            with open(config_path) as config_file:
-                config = json.load(config_file)
-                self.site = Papery(config)
+            if os.path.exists('config.yaml') or os.path.exists('.config.yaml'):
+                if os.path.exists('config.yaml'):
+                  config_path = 'config.yaml'
```
Comment (federkasten) [inline papery/engine.py:75]:
```suggestion
                    config_path = 'config.yaml'
```

**Review by federkasten — APPROVED** [review]
LGTM :+1:  I commented on some minor points.

File: papery/rendering.py [hunk 4]
```diff
@@ -184,7 +184,10 @@ def _render_pages(self):
 
         for page, render_vars in self._targets.items():
             page_base, _ = os.path.splitext(page)
-            info = page_base + ".json"
+            if os.path.exists(page_base + ".yaml"):
+              info = page_base + ".yaml"
+            else:
+              info = page_base + ".json"
```
Comment (federkasten) [inline papery/rendering.py:190]:
```suggestion
            if os.path.exists(page_base + ".yaml"):
                info = page_base + ".yaml"
            else:
                info = page_base + ".json"
```

**Review by federkasten — APPROVED** [review]

**Pull request merged**